### ISSUE TITLE
Complete GROQ syntax tests and documentation

### ISSUE THREAD

**Issue by erikgrinaker** [sanity-io/gradient#550]
We need complete and comprehensive test coverage of all documented GROQ syntax. This should be implemented using the current integration test suite, and then ported to the improved E2E test suite in #525 once it is ready. The primary objective is syntax testing - implementation tests, such as ES pagination, optimization codepaths, access control, performance regressions etc. will be postponed to #525 as well.

In parallel with this, we should write complete and comprehensive reference documentation for GROQ.

**Comment by erikgrinaker** [comment]
We should now have fairly complete GROQ integration tests at https://github.com/sanity-io/gradient/tree/master/integration-tests/reference_queries

They're not as systematically complete as I'd like (for this we'll need to generate test case permutations programmatically, see #525), but I believe they cover most of what's necessary. Please let me know if you feel like something is missing.

Docs are still WIP.
